api_docgen/odoc/Makefile

```
@@ -0,0 +1,188 @@
+
+$(ALL_PAGE_TEXT:%=build/%.odocl):%.odocl:%.odoc \
+	| $(ALL_PAGED_DOC:%=build/%.odoc)
+	$(odoc) link -I build/libref -I build/compilerlibref $<
+
+# Rules for all three backends:
+
+ALL_HTML = $(ALL_PAGED_DOC:%=build/%.html.stamp)
+ALL_MAN = $(ALL_PAGED_DOC:%=build/%.3o.stamp)
+ALL_LATEX = $(ALL_PAGED_DOC:%=build/%.tex.stamp)
+
+build/libref/stdlib.html.stamp: $(STDLIB:%=build/libref/%.odocl) | build/libref
+build/libref/stdlib.3o.stamp: $(STDLIB:%=build/libref/%.odocl) | build/libref
+build/libref/stdlib.tex.stamp: $(STDLIB:%=build/libref/%.odocl) | build/libref
+
+man: $(ALL_MAN)
+html: $(ALL_HTML) build/html/odoc.css
+html: build/libref/index.html.stamp build/compilerlibref/index.html.stamp
+
+# Html rules
+$(ALL_HTML): %.html.stamp: %.odocl | build/html
+	$(odoc) html-generate --output-dir build/html  $<
+	touch $@
+
+build/html/odoc.css: | build/html
+	$(odoc) support-files --output-dir build/html
+
+$(build/libref.html.stamp build/compilerlibref.html.stamp):
+%.html.stamp: %.mld | build/
+	$(odoc) html-generate --output-dir build/html  $<
+	touch $@
+
+# Html indexes for the api documentation
+
+# The stdlib index is generated from the list of stdlib modules.
+stdlib_INDEX=\
+	$(foreach m,$(stdlib_UNPREFIXED),$(call stdlib_prefix,$m))\
+	$(call capitalize, $(otherlibref))
+build/libref.mld:
+	echo {0 OCaml standard library} {!modules:$(stdlib_INDEX)} > $@
+
+build/libref/index.html.stamp: $(ALL_HTML) build/libref.mld | build/libref
+	$(odoc) compile --package libref build/libref.mld
+	$(odoc) link -I build/libref build/page-libref.odoc
+	$(odoc) html-generate build/page-libref.odocl --output-dir build/html
+	mv build/html/libref/libref.html build/html/libref/index.html
+	touch $@
+
+build/compilerlibref/index.html.stamp: $(ALL_HTML) \
+  build/compilerlibref/page-Compiler_libs.html.stamp | build/compilerlibref
+	cp build/html/compilerlibref/Compiler_libs.html \
+           build/html/compilerlibref/index.html
+	touch $@
+
+# Latex rules
+
+latex: $(ALL_LATEX)
+
+
+build/latex/alldoc.pdf: $(ALL_LATEX) build/latex/alldoc.tex \
+build/latex/stdlib_input.tex build/latex/compilerlibs_input.tex \
+| build/latex
+	cd build/latex && pdflatex alldoc.tex
+	cd build/latex && pdflatex alldoc.tex
+
+# We include children pages directly except for the root Stdlib module
+NOT_STDLIB=$(filter-out libref/stdlib,$(ALL_PAGED_DOC))
+$(NOT_STDLIB:%=build/%.tex.stamp):\
+build/%.tex.stamp: build/%.odocl | build/
+	$(odoc) latex-generate --with-children=true --output-dir build/latex $<
+	touch $@
+
+# Stdlib latex page: we manually integrate stdlib module
+build/libref/stdlib.tex.stamp: build/libref/stdlib.odocl | build/libref
+	$(odoc) latex-generate --with-children=false --output-dir build/latex $<
+	touch $@
+
+# Man pages
+$(ALL_PAGED_DOC:%=build/%.3o.stamp):build/%.3o.stamp:build/%.odocl | build/
+	$(odoc) man-generate --output-dir build/man  $<
+	touch $@
+
+# Man pages are the only installed documentation
+INSTALL_MANODIR=$(INSTALL_MANDIR)/man3
```

```suggestion
INSTALL_MAN3DIR=$(INSTALL_MANDIR)/man3
```
? The previous name looked like a typo?

---

Maybe, but this would be a mostly unrelated change?

---

Sort of - though this _addition_ looked like a typo when I reviewed it (even if the addition is the result of moving code from somewhere else)